emulator/iSkipper/iClickerEmulator.h

```diff
@@ -23,6 +23,44 @@ enum iClickerAnswer : uint8_t
     NUM_ANSWER_CHOICES,
 };
 
+struct QuestionMode 
+{
+    // magic bytes used in the welcome packet for this
+    // question mode
+    uint8_t welcome_bytes[2];
+};
+
+namespace QuestionModes {
+    const QuestionMode MULTIPLE_CHOICE = 
+    {
+        {0x92, 0x62}
+    };
+
+    const QuestionMode NUMERIC =
+    {
+        {0x93, 0x63}
+    };
+
+    const QuestionMode ALPHANUMERIC =
+    {
+        {0x9B, 0x6B}
+    };
+
+    const QuestionMode MULTIPLE_NUMERIC =
+    {
+        {0x94, 0x64}
+    };
+
+    const QuestionMode MULTIPLE_ALPHANUMERIC =
+    {
+        {0x9C, 0x6C}
+    };
+};
+
+#define WELCOME_RECV_SYNC_ADDR_LEN 6
+const uint8_t WELCOME_RECV_SYNC_ADDR[WELCOME_RECV_SYNC_ADDR_LEN] = 
+    {0x55, 0x55, 0x55, 0x36, 0x36, 0x36};
```

Comment:
I imagine you see the base station's firmware transmitting this leading `0x55` . Remember `0x55 = 0b01010101`.
This is used to help the receiver learn the channel, not to actually try to trigger on using the sync address functionality. You can not reliable assume it will be received `{0x55, 0x55, 0x55}`.  The sync address should just be `{0x36, 0x36, 0x36}`. Also remember when calling `_radio.send(msg)` the RFM69 is automatically creating the packet to transmit as `{0x55, 0x55, ..., 0x55} + sync_addr + msg`, as it automatically prepends the `0x55` preamble and the sync address currently configured.

This actually bring up a problem with the way thehttps://github.com/wizard97/iSkipper/pull/12. Trying to receive the ack by triggering using the sync address functionality on the preamble ({0x55, 0x55, 0x55}) likely will not work reliably in certain situations. I think we should use the first two bytes of the encodedID instead.

Reply:
> as it automatically prepends the 0x55 preamble

The RFM69HCW's datasheet is really confusing me here, the only configurable part of the preamble seems to be `RegPreambleMsb` and `RegPreambleLsb` which defaults to `0x03`, so is the preamble hardcoded to just be `0x55`? Is this just an industry convention or something?

> Trying to receive the ack by triggering using the sync address functionality on the preamble ({0x55, 0x55, 0x55}) likely will not work reliably in certain situations. I think we should use the first two bytes of the encodedID instead.

The datasheet does mention the following:

* A preamble (0x55 or 0xAA) of 12 bits is required for synchronization (from the RxReady interrupt)

So only 1.5 bytes are required to trigger the preamble detecting part of the RFM69 and the rest should be safe to use as a sync word? Also won't this present a problem for promiscuously snooping on all ACKs? What do we use as the sync word there?

Reply:
> the preamble hardcoded to just be 0x55? Is this just an industry convention or something?
Yeah, the datasheet is a bit confusing... Yeah, the preamble is an alternating series of 1 and 0s. So both 0b01010101 (0x55) as well as 0b10101010 (0xaa) would work.

> So only 1.5 bytes are required to trigger the preamble detecting part of the RFM69 and the rest should be safe to use as a sync word? 

1.5 bytes is only the minimum. I'm not sure if with this particular chipset a longer preamble helps after the first 1.5 bytes. After reading the datasheet more carefully it looks like it is mostly being used to lock on to the bitrate of the transmission. Remember this is supposed to be a general purpose FSK transceivers, so they designed it to be compatible with other transceivers. Some more advanced FSK receivers could use this preamble to do additional things.
Most likely. 

> Also won't this present a problem for promiscuously snooping on all ACKs? What do we use as the sync word there?
Yeah, this is a problem with trying to support promisoucMode for acks. See https://github.com/wizard97/iSkipper/issues/13